interfaces/ifacetest/backendtest.go

```diff
@@ -147,10 +147,15 @@ slots:
 // Support code for tests
 
 // InstallSnap "installs" a snap from YAML.
-func (s *BackendSuite) InstallSnap(c *C, opts interfaces.ConfinementOptions, snapYaml string, revision int) *snap.Info {
+func (s *BackendSuite) InstallSnap(c *C, opts interfaces.ConfinementOptions, instanceName, snapYaml string, revision int) *snap.Info {
 	snapInfo := snaptest.MockInfo(c, snapYaml, &snap.SideInfo{
 		Revision: snap.R(revision),
 	})
+
+	_, instanceKey := snap.SplitInstanceName(instanceName)
```

Comment:
Per review in the prior PR, can't we make the name default to the one in the snapInfo, so we don't have to repeat it all the time?  This way most cases would become an empty string instead of repeating the same name which is then ignored here. Also, would be good to validate that when provided the name matches the one in the info instead of ignoring it, to prevent awkwardness.

Reply:
I recall an argument, in a PR adding something similar in other package, that making this explicit was better. Added anyway. Validation was already there.